### PR TITLE
DIP-273 Content Addressing

### PR DESCRIPTION
**Pull request by wesbiggs**
Problem
=======
DSNP should provide affordances for finding content that do not rely on a single DNS-based point of failure for content hosting.
More in the original discussion: #273 

Solution
========
Enhance the specification to treat URLs in Announcements as suggestions rather than canonical locations for content.
Provide a simple and well-specified set of hashes and encodings that can be used consistently throughout the protocol.
Use IPFS CIDv1 specifically for locating profiles.

Change summary:
---------------
* Broaden ProfileResource definition so that different types of profile resources can use different, possibly distributed, file systems via a generic `contentAddress` field
* Simplify multihashes to use base32 encoding only and sha2-256 or blake3 as the hashing algorithm
* Update various example hashes in line with this
* Update to pre-1.3.0 versioning and sync prerelease changelogs for other recent additions to the spec
* Change spec language to expand on how to treat content hash + URL pairs.
* Announcements use the same base32 encoded multihash for the `contentHash` and `targetContentHash` fields.

### CODE REVIEW

File: pages/DSNP/Identifiers.md [hunk 1]
```diff
@@ -69,9 +70,9 @@ In order for DSNP applications to interoperate, the required functionality is li
 
 - CID version: MUST be version 1, in order to distinguish CIDs from simple multihash values in situations where either may be used
 - Hash algorithm: MUST be `sha2-256` or `blake2b-256`
-- Encoding: MUST be `base58btc` or `base32`
 - Codec: MUST be `dag-pb` for data 256*1024 bytes or longer; `raw` for data less than 256*1024 bytes
```
Comment (audreyt) [inline pages/DSNP/Identifiers.md:73]: Adding backslash before `*` helps markdown to disambiguate:

- Codec: MUST be `dag-pb` for data 256\*1024 bytes or longer; `raw` for data less than 256\*1024 bytes

**Review by wilwade — APPROVED** [review]
Looks good to me!

**Review by shannonwells — APPROVED** [review]
Read through the referenced Issue thread, read through the changes, looks ok by me.

**Review by saraswatpuneet — APPROVED** [review]

File: pages/DSNP/BatchPublications.md [hunk 2]
```diff
@@ -43,13 +43,13 @@ See also [Announcement Types](Announcements.md).
...
-| fromId | `BYTE_ARRAY` |
-| inReplyTo | `BYTE_ARRAY` |
-| objectId | `BYTE_ARRAY` |
+| Column | Primitive Type | Logical Type | Converted Type (deprecated) |
+| ------ | ---- | ---- | --- |
+| contentHash | `BYTE_ARRAY` | `STRING` | `UTF8` |
+| emoji | `BYTE_ARRAY` | `STRING` | `UTF8` |
+| fromId | `INT64` | `INT(64, false)` | `UINT_64` |
+| inReplyTo | `BYTE_ARRAY` | `STRING` | `UTF8` |
+| targetContentHash | `BYTE_ARRAY` | `STRING` | `UTF8` |
```
Comment (saraswatpuneet) [inline pages/DSNP/BatchPublications.md:52]: Note: Confirming on gateway side this is already updated

**Review by shannonwells — APPROVED** [review]
Re-approved

**Review by shannonwells — APPROVED** [review]
This time for sure

**Pull request merged**